metecho/api/models.py

```diff
@@ -109,38 +109,18 @@ def notify(self, subject, body):
         )
 
     def queue_refresh_repositories(self):
-        """Queue a job to refresh repositories unless we're already doing so"""
         from .jobs import refresh_github_repositories_for_user_job
 
         if not self.currently_fetching_repos:
             self.currently_fetching_repos = True
             self.save()
             refresh_github_repositories_for_user_job.delay(self)
 
-    def refresh_repositories(self):
-        try:
-            repos = gh.get_all_org_repos(self)
-            with transaction.atomic():
-                GitHubRepository.objects.filter(user=self).delete()
-                GitHubRepository.objects.bulk_create(
-                    [
-                        GitHubRepository(
-                            user=self,
-                            repo_id=repo.id,
-                            repo_url=repo.html_url,
-                            permissions=repo.permissions,
-                        )
-                        for repo in repos
-                    ]
-                )
-        finally:
-            self.refresh_from_db()
-            self.currently_fetching_repos = False
-            self.save()
-            self.notify_repositories_updated()
```

Comment:
@jgerigmeyer @davisagli it turns out when refreshing user repos we didn't have any error handling in place. This makes the `currently_fetching_repos` flag get stuck if the job is interrupted. I have refactored the the job to be in line with how we handle other async tasks and this time I think we should get the flag set to `False` even if the worker is shut down.

I'll continue with the other changes listed on the card but just wanted to note this